features/project/source/browse_files.feature

```diff
@@ -33,6 +33,13 @@ Feature: Project Source Browse Files
     And I click on "Commit Changes"
     Then I am redirected to the new file
     And I should see its new content
+    
+  @javascript
+  Scenario: I can upload file and commit
+    Given I click on "new file" link in repo
+    Then I can see new file page
+    And I can see "upload existing one"
+    And I click on "upload existing one"
```

Comment:
@stanhu about the issue 6, I checked that the create_service.rb's checking will work when we are not at the head tree. If I uploaded one file in the sub-folder, then it will work. 

For example,
1). No error prompt when we in head tree 
![tver_head01](https://cloud.githubusercontent.com/assets/7370334/9815767/d5a6c860-5898-11e5-8cfb-f829ba7f0546.png)
![tver_head02](https://cloud.githubusercontent.com/assets/7370334/9815769/d5aa634e-5898-11e5-868f-52e3dc2c9f91.png)
![tver_head03](https://cloud.githubusercontent.com/assets/7370334/9815768/d5a9b96c-5898-11e5-8958-8ff255c0d347.png)

2). error prompt when we are in sub-folder
![tver_sub01](https://cloud.githubusercontent.com/assets/7370334/9815786/f8dff4b4-5898-11e5-9558-5a983de17e7c.png)
![tver_sub02](https://cloud.githubusercontent.com/assets/7370334/9815788/f8e424c6-5898-11e5-88a7-6bd729692853.png)
![tver_sub03](https://cloud.githubusercontent.com/assets/7370334/9815787/f8e2e6ec-5898-11e5-8500-8cda16e8e790.png)